source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -421,6 +425,59 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
+                if (!TrySetAws())
+                {
+                    log.Warn("Could not find the aws cli, falling back to the aws-iam-authenticator.");
+                }
+                else
+                {
+                    try
+                    {
+                        var region = Regex.Match(clusterUrl, @"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d", RegexOptions.IgnoreCase).Value;
```

Comment:
Looks like the service url for EKS has a fixed format: https://docs.aws.amazon.com/general/latest/gr/eks.html

Should we use the format to get the region section (for example, trimming the `.amazonaws.com` from the end and so on), rather than matching with the regex? The risk for using regex is that amazon may expand to other regions in the future. In that case we need to come back add modify the regex pattern, which could be error-prone